### ISSUE TITLE
[SALESFORCE] Scheduler Pro / Gantt task editor loses focus and cancels on click

### ISSUE THREAD

**Issue by ghulamghousdev** [bryntum/support#9993]
[Forum post](https://forum.bryntum.com/viewtopic.php?f=51&t=30364&p=154127#p154127)

Hi there!

I think I may have found a bug in how Scheduler Pro and Gantt event editors handle focus.

Open Scheduler Pro or Gantt demo tab and double-click on a task to edit. When the editor appears, try changing any value (or even switch to a different editor tab) using mouse. The editor disappears immediately as if the click was outside of the dialog window. Same thing happens after pressing the Tab key once the editor opens.

Event editor in the "regular" Scheduler (not Pro) works just fine, using both mouse and keyboard.

At first I thought this was a glitch, but then I confirmed that it is consistently happening, including in your Salesforce demo environment. This behaviour is new to v6, it was working fine in v5.

https://github.com/user-attachments/assets/44560cd5-baa6-4957-bce1-74cc28d08e9a

**Comment by amurashincertinia** [comment]
This override seem to have worked:

```js
import { GlobalEvents, Override } from '@bryntum/core-thin';

{
    class GlobalEventsOverride {
        static get target() {
            return {
                class: GlobalEvents,
            };
        }

        static setupFocusListenersOnce(rootElement, EventHelper) {
            // Shouldn't attach global listeners to the body in Salesforce

            if (rootElement !== document.body) {
                this._overridden.setupFocusListenersOnce.apply(this, arguments);
            }
        }
    }

    Override.apply(GlobalEventsOverride);
}
```

What was happening is that initially all calls to `setupFocusListenersOnce ` were made with the `rootElement` being the `div` with `lwc:dom="manual" attribute which hosts the Bryntum component. But when a Task Editor was created in Scheduler Pro or Gantt, the `rootElement` was `document.body` (just like outside of Salesforce environment). This caused global listeners to be called twice for each focus event, which led to those issues.

By explicitly forbidding global listeners on the `body` element in Salesforce environment, we are limiting those only to individual host elements and eliminating duplicate focus event handling. The issue goes away.

**Comment by amurashincertinia** [comment]
☝️ FYI @chuckn0rris.

**Comment by amurashincertinia** [comment]
Although a better fix would probably be to ensure that the `Widget.rootElement` property getter always returns the correct host element, rather than `document.body`, when in the Salesforce environment.

**Comment by chuckn0rris** [comment]
Not sure we should override it globally at this point, I prefer to stay with a fix for `setupFocusListenersOnce` until we got a test coverage which is coming.

**Comment by amurashincertinia** [comment]
Happy to help!

I did some digging and found that the `CalendarEditor` widget in Scheduler Pro has a default config for its `rootElement`, which is explicitly set to `document.body`. This forces all of its child elements to also use `document.body` as the `rootElement`, which is why `setupFocusListenersOnce` is called with the body element as root when a Task Editor is crated.

Removing the default value seems to be fixing the issue without any overrides necessary, and the component continues to work outside of Salesforce just fine. But I did not do enough testing to confirm that everything is perfect, so this "catch all" override might be an OK solution short term.
